packages/components/src/internal/components/settings/NameIdSettings.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { FC, useCallback, useEffect, useReducer } from 'react';
+
+import { PermissionTypes } from '@labkey/api';
+import { Button, Checkbox, FormControl } from 'react-bootstrap';
+
+import { Alert, ConfirmModal, LabelHelpTip, LoadingSpinner, RequiresPermission } from '../../..';
+
+import { sampleManagerIsPrimaryApp } from '../../app/utils';
+
+import { loadNameExpressionOptions, saveNameExpressionOptions } from './actions';
+
+export const NameIdSettings = () => {
+    return (
+        <RequiresPermission perms={PermissionTypes.Admin}>
+            <NameIdSettingsForm
+                saveNameExpressionOptions={saveNameExpressionOptions}
+                loadNameExpressionOptions={loadNameExpressionOptions}
+            />
+        </RequiresPermission>
+    );
+};
+
+interface Props {
+    loadNameExpressionOptions: () => Promise<{ prefix: string; allowUserSpecifiedNames: boolean }>;
+    saveNameExpressionOptions: (key: string, value: string | boolean) => Promise<void>;
+}
+
+interface State {
+    error: string;
+    loading: boolean;
+    prefix: string;
+    savingPrefix: boolean;
+    allowUserSpecifiedNames: boolean;
+    savingAllowUserSpecifiedNames: boolean;
+    confirmModalOpen: boolean;
+}
+
+const initialState: State = {
+    error: undefined,
+    loading: true,
+    prefix: '',
+    savingPrefix: false,
+    confirmModalOpen: false,
+    allowUserSpecifiedNames: false,
+    savingAllowUserSpecifiedNames: false,
+};
+export const NameIdSettingsForm: FC<Props> = props => {
+    const { loadNameExpressionOptions, saveNameExpressionOptions } = props;
+    const [state, setState] = useReducer(
+        (currentState: State, newState: Partial<State>): State => ({ ...currentState, ...newState }),
+        initialState
+    );
+
+    const {
+        loading,
+        savingAllowUserSpecifiedNames,
+        allowUserSpecifiedNames,
+        prefix,
+        savingPrefix,
+        confirmModalOpen,
+        error,
+    } = state;
+
+    const initialize = async () => {
+        const payload = await loadNameExpressionOptions();
+        setState({
+            prefix: payload.prefix ?? '',
+            allowUserSpecifiedNames: payload.allowUserSpecifiedNames,
+            loading: false,
+        });
+    };
```

Comment:
```suggestion
    const initialize = async (): Promise<void> => {
        try {
            const payload = await loadNameExpressionOptions();
            setState({
                prefix: payload.prefix ?? '',
                allowUserSpecifiedNames: payload.allowUserSpecifiedNames,
                loading: false,
            });
        } catch (err) {
            setState({ error: err, loading: false });
        }
    };
```

`async/await` is just a shorthand way to create a promise that also allows you to write your code without callbacks (which is the big advantage to using async/await). This means that the type for an `async` void function is `Promise<void>`. To catch errors you use a traditional try/catch.